### ISSUE TITLE
False positive: UnusedVariable for record components that are not read directly

### ISSUE THREAD

**Issue by anthonyvdotbe** [google/error-prone#1648]
### Description of the problem / feature request:

A false positive for `UnusedVariable` is reported for all record components that are only read through methods.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

````java
class Foo {

    record Bar(int v) {}

    int baz() {
        return new Bar(42).v();
    }

}
````

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No.

**Comment by ferozco** [comment]
We built a [stricter](https://github.com/palantir/gradle-baseline/blob/develop/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/StrictUnusedVariable.java) version of `UnusedVariable` in [gradle-baseline](https://github.com/palantir/gradle-baseline/) and we were able to make the rule compatible with records with this [change](https://github.com/palantir/gradle-baseline/pull/1412) which should be directly applicable to `UnusedVariable`

**Comment by codylerum** [comment]
Same hit on this 

Just the most simple record will fail

```
public record AgentAndCount(String agent, long count) {
}
```

```
Warning:  /home/runner/work/oss/oss/src/main/java/co/foo/oss/agent/AgentAndCount.java:[3,34] [UnusedVariable] The field 'agent' is never read.
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean 'public record AgentAndCount(, long count) {'?
Warning:  /home/runner/work/oss/oss/src/main/java/co/foo/oss/agent/AgentAndCount.java:[3,46] [UnusedVariable] The field 'count' is never read.
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean 'public record AgentAndCount(String agent) {'?
```

**Comment by tschniewind** [comment]
Still failes if record is private as in
``
class Foo {

  private record Bar(int v) {
  }

  int baz() {
    return new Bar(42).v();
  }

}
``

**Comment by nvervelle** [comment]
> Please reopen, cause it still failes if record is private as in

This is why #2713 was opened :-)
